lib/rpmsg/rpmsg_virtio.c

```diff
@@ -587,12 +587,10 @@ int rpmsg_init_vdev(struct rpmsg_virtio_device *rvdev,
 	 * service announcement feature.
 	 */
 	if ((dev_features & (1 << VIRTIO_RPMSG_F_NS))) {
-		if (!rpmsg_create_ept(rdev, "NS", RPMSG_NS_EPT_ADDR,
-				      RPMSG_NS_EPT_ADDR,
-				      rpmsg_virtio_ns_callback,
-				      NULL))
-			return RPMSG_ERR_NO_MEM;
-
+		rpmsg_init_ept(&rdev->ns_ept, "NS",
```

Comment:
Why change this from being dynamically allocated to static?

Reply:
This is to allow static allocation without the need to link with memory allocation function

Reply:
Not sure how much that helps, since we still have allocation happening in the ns_callback.  So not sure if I see much value in making this static.

Reply:
sth like macro?